presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java

```diff
@@ -257,15 +293,21 @@ private static ConnectorPageSource createParquetPageSource(
                             predicateMatches(parquetPredicate, block, dataSource, descriptorsByPath, parquetTupleDomain, columnIndexStore, false, Optional.of(session.getWarningCollector()))) {
                         blocks.add(block);
                         blockIndexStores.add(columnIndexStore.orElse(null));
+                        blockStarts.add(nextStart);
+                        if (!startRowPosition.isPresent()) {
```

Comment:
`startRowPosition`.isEmpty()

Reply:
The `Optional` class of jdk 1.8 does not have an `isEmpty` method, which was introduced in jdk11

Reply:
Oops... Thanks for the clarification.